### PR TITLE
Allow more than 2 axis in `F.normalize`

### PR DESCRIPTION
**Pull request by niboshi**
As the use of `np.linalg.norm` has been eliminated in #4763, we no longer have to limit the number of axis in `F.normalize`.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/normalization_tests/test_l2_normalization.py [hunk 1]
```diff
@@ -21,6 +21,8 @@
         {'shape': (4, 3, 2, 5), 'axis': 2},
         {'shape': (4, 3, 2, 5), 'axis': 3},
         {'shape': (4, 3, 2), 'axis': (0, 1)},
+        {'shape': (4, 3, 2, 4, 3, 2, 2), 'axis': (1, 4, 3, 6)},
+        {'shape': (0, 2), 'axis': 1},
```
Comment (toslunar) [inline tests/chainer_tests/functions_tests/normalization_tests/test_l2_normalization.py]: `NormalizeL2.forward` needs `force_array` in order to support `len(axis) == 0`.  (Test `{'shape': (), 'axis': ()},`.)

File: chainer/functions/normalization/l2_normalization.py [hunk 2]
```diff
@@ -59,13 +57,14 @@ def normalize(x, eps=1e-5, axis=1):
     This function implements L2 normalization on a vector along the given axis.
     No reduction is done along the normalization axis.
 
-    In the case when :obj:`axis=1` and :math:`x` is a vector of dimension
-    :math:`(N, K)`, where :math:`N` and :math:`K` denote mini-batch size and
-    the dimension of the input variable, this function computes an output
-    vector :math:`y` by the following equation:
+    This function computes an output vector :math:`y` by the following
+    equation:
```
Comment (toslunar) [inline chainer/functions/normalization/l2_normalization.py]: The original paragraph showed the most typical example.  I'd like to preserve it somewhere in this docs.

File: tests/chainer_tests/functions_tests/normalization_tests/test_l2_normalization.py (hunk 1 above)
Comment (niboshi) [reply tests/chainer_tests/functions_tests/normalization_tests/test_l2_normalization.py]: Thank you for pointing out. Fixed.
(Casting `eps` is needed to support ()-shape.)

File: chainer/functions/normalization/l2_normalization.py (hunk 2 above)
Comment (niboshi) [reply chainer/functions/normalization/l2_normalization.py]: I reverted to the original formula, and updated some notation to make it clear `x_i` and `y_j` are vectors.
